elm/src/Main.elm

```diff
@@ -0,0 +1,373 @@
+port module Main exposing (..)
+
+import Date exposing (..)
+import Date.Extra exposing (..)
+import Dom.Scroll exposing (..)
+import Html exposing (..)
+import Html.Attributes as HA exposing (..)
+import Html.Events exposing (..)
+import Http exposing (..)
+import Json.Decode exposing (..)
+import Json.Decode.Pipeline as JPipe exposing (..)
+import Json.Encode exposing (Value)
+import Task exposing (..)
+import Time exposing (Time)
+import Window exposing (..)
+
+
+-- Maybe String because the name is initialised from localstorage (if it exists)
+
+
+main : Program (Maybe String) Model Msg
+main =
+    Html.programWithFlags
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { name : String
+    , messages : List Message
+    , messageInput : MessageInput
+    , nameInput : String
+    , windowWidth : Int
+    }
+
+
+
+-- Match message types to the backend
+
+
+type alias Message =
+    { n : String
+    , t : Time
+    , m : String
+    }
+
+
+
+-- Add a placeholder to messageInput so empty message submits prompt the user
```

Comment:
Yes we both do, are you referring to the line breaks with between the comments?
I just reduced all line breaks to 1, and on save elm format reformatted to the same as the quoted block here.